mithril-client-wasm/www-test/index.js

```diff
@@ -0,0 +1,159 @@
+import initMithrilClient, {
+  MithrilClient,
+} from "@mithril-dev/mithril-client-wasm"
+
+function display_test_result_in_dom(step_number, step_name, result, error) {
+  let div = document.createElement("div")
+  div.id = step_name
+  div.title = result
+  div.innerHTML = `Result test n°${step_number}: ${result}; function_name: ${step_name}${
+    error ? `; reason: ${error}` : ""
+  }`
+  document.body.appendChild(div)
+}
+
+function handle_error(step_number, step_name, error) {
+  display_test_result_in_dom(step_number, step_name, "FAILED", error)
+  console.error(`Error at step ${step_number} (${step_name}):`, error)
+  add_finished_div()
+  throw new Error(
+    `Stopping script due to error at step ${step_number}: ${error}`
+  )
+}
+
+function add_finished_div() {
+  let div = document.createElement("div")
+  div.id = "tests_finished"
+  document.body.appendChild(div)
+}
+
+await initMithrilClient()
+const aggregator_endpoint = process.env.AGGREGATOR_ENDPOINT
+const genesis_verification_key = process.env.GENESIS_VERIFICATION_KEY
+let client
+let test_name
+let test_number = 1
+
+// Test 'MithrilClient constructor'
+try {
```

Comment:
You can factor those try/catch using a function, something like this should work:

```js
function test(name, step,  fun) {
  try {
    fun();
    display_test_result_in_dom(step, name, "OK");
  } catch (error) {
    handle_error(test_number, test_name, error);
  }
}

let client;
let step = 1;

test("constructor", step, () => {
  client = new MithrilClient(aggregator_endpoint, genesis_verification_key);
});

step++;
test("list_snapshots", step, () => {
  snapshots = await client.list_snapshots();
  console.log("snapshots", snapshots);
});
```

Reply:
Thanks 🙏